lib/modules/versioning/debian/index.spec.ts

```diff
@@ -0,0 +1,377 @@
+    ${'experimental'} | ${false}
+  `('isValid("$version") === $expected', ({ version, expected }) => {
+    expect(debian.isValid(version)).toBe(expected);
+  });
+
+  it.each`
+    version           | range        | expected
+    ${undefined}      | ${undefined} | ${false}
+    ${null}           | ${undefined} | ${false}
+    ${''}             | ${undefined} | ${false}
+    ${'7'}            | ${undefined} | ${true}
+    ${'11'}           | ${undefined} | ${true}
+    ${'12'}           | ${undefined} | ${false}
+    ${'stable'}       | ${undefined} | ${true}
+    ${'oldstable'}    | ${undefined} | ${true}
+    ${'oldoldstable'} | ${undefined} | ${true}
+    ${'wheezy'}       | ${undefined} | ${true}
+    ${'bullseye'}     | ${undefined} | ${true}
+    ${'bookworm'}     | ${undefined} | ${false}
+  `(
+    'isCompatible("$version") === $expected',
+    ({ version, range, expected }) => {
+      expect(debian.isCompatible(version, range)).toBe(expected);
+    }
+  );
+
+  it.each`
+    version      | expected
+    ${undefined} | ${false}
+    ${null}      | ${false}
+    ${''}        | ${false}
+    ${'6'}       | ${true}
+    ${'>=6'}     | ${false}
+  `('isSingleVersion("$version") === $expected', ({ version, expected }) => {
+    expect(debian.isSingleVersion(version)).toBe(expected);
+  });
+
+  it.each`
+    version           | expected
+    ${undefined}      | ${false}
+    ${null}           | ${false}
+    ${''}             | ${false}
+    ${'buzz'}         | ${false}
+    ${'rex'}          | ${false}
+    ${'bo'}           | ${false}
+    ${'hamm'}         | ${false}
+    ${'slink'}        | ${false}
+    ${'potato'}       | ${false}
+    ${'woody'}        | ${false}
+    ${'sarge'}        | ${false}
+    ${'etch'}         | ${false}
+    ${'lenny'}        | ${false}
+    ${'squeeze'}      | ${false}
+    ${'wheezy'}       | ${false}
+    ${'jessie'}       | ${false}
+    ${'stretch'}      | ${true}
+    ${'buster'}       | ${true}
+    ${'bullseye'}     | ${true}
+    ${'bookworm'}     | ${false}
+    ${'trixie'}       | ${false}
+    ${'sid'}          | ${false}
+    ${'1.1'}          | ${false}
+    ${'1.2'}          | ${false}
+    ${'1.3'}          | ${false}
+    ${'2'}            | ${false}
+    ${'2.1'}          | ${false}
+    ${'2.2'}          | ${false}
+    ${'3'}            | ${false}
+    ${'4'}            | ${false}
+    ${'5'}            | ${false}
+    ${'6'}            | ${false}
+    ${'7'}            | ${false}
+    ${'8'}            | ${false}
+    ${'9'}            | ${true}
+    ${'10'}           | ${true}
+    ${'11'}           | ${true}
+    ${'12'}           | ${false}
+    ${'13'}           | ${false}
+    ${'sid'}          | ${false}
+    ${'experimental'} | ${false}
+    ${'stable'}       | ${true}
+    ${'oldstable'}    | ${true}
+    ${'oldoldstable'} | ${true}
+  `('isStable("$version") === $expected', ({ version, expected }) => {
+    expect(debian.isStable(version)).toBe(expected);
```

Comment:
We need to mock time here too, like we do in ubuntu.

Reply:
did that but didnt push yet. ive moved all the time handling into distro.ts.
this might need another refactor PR as its affecting ubuntu as well, what do u think ?

Reply:
yes, please do so